src/qml/filters/gpstext/ui.qml

```diff
@@ -0,0 +1,815 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+import QtQuick.Dialogs 1.2
+import QtQuick.Window 2.1
+import Shotcut.Controls 1.0 as Shotcut
+import org.shotcut.qml 1.0 as Shotcut
+import QtQml.Models 2.12
+
+Item {
+	id: gpsTextRoot
+	width: 300
+    height: 800
+	
+	property url settingsOpenPath: 'file:///' + settings.openPath
+	Shotcut.File { id: gpsFile }
+	
+	// This signal is used to workaround context properties not available in
+	// the FileDialog onAccepted signal handler on Qt 5.5.
+	signal fileOpened(string path)
+	onFileOpened: settings.openPath = path
+	
+	Component.onCompleted: {
+		var resource = filter.get('gps.file')
+		gpsFile.url = resource
+		
+		filter.blockSignals = true
+	        
+		filter.set(textFilterUi.middleValue, Qt.rect(0, 0, profile.width, profile.height))
+		filter.set(textFilterUi.startValue, Qt.rect(0, 0, profile.width, profile.height))
+		filter.set(textFilterUi.endValue, Qt.rect(0, 0, profile.width, profile.height))
+		if (filter.isNew) {
+			var presetParams = preset.parameters.slice()
+			var index = presetParams.indexOf('argument')
+			if (index > -1)
+				presetParams.splice(index, 1)
+
+			if (application.OS === 'Windows')
+				filter.set('family', 'Verdana')
+			filter.set('fgcolour', '#ffffffff')
+			filter.set('bgcolour', '#00000000')
+			filter.set('olcolour', '#aa000000')
+			filter.set('outline', 3)
+			filter.set('weight', 10 * Font.Normal)
+			filter.set('style', 'normal')
+			filter.set(textFilterUi.useFontSizeProperty, false)
+			filter.set('size', profile.height)
+			
+			// Add default preset.
+			filter.set(textFilterUi.valignProperty, 'bottom')
+			filter.set(textFilterUi.halignProperty, 'left')
+			filter.set(textFilterUi.rectProperty, '10%/10%:80%x80%')
+			filter.savePreset(presetParams)
+		} else {
+			filter.set(textFilterUi.middleValue, filter.getRect(textFilterUi.rectProperty, filter.animateIn + 1))
+			if (filter.animateIn > 0)
+				filter.set(textFilterUi.startValue, filter.getRect(textFilterUi.rectProperty, 0))
+			if (filter.animateOut > 0)
+				filter.set(textFilterUi.endValue, filter.getRect(textFilterUi.rectProperty, filter.duration - 1))
+		}
+		filter.blockSignals = false
+		setControls()
+		
+		if (filter.isNew)
+			filter.set(textFilterUi.rectProperty, filter.getRect(textFilterUi.rectProperty))
+			
+		if (gpsFile.exists()) {
```

Comment:
These should be handled in setControls()

Reply:
Done in latest commit.